### DISCUSSION TITLE
Unable to connect on configuring "graphql-ws" in Nestjs

### DISCUSSION THREAD

**Discussion by nasim-90** [enisdenjo/graphql-ws#605] [Q&A]
## After installation of `graphql-ws` when configuring app module 

```
subscriptions: {
      'graphql-ws': {
        path: '/graphql',
        onConnect: (connectionParams: any, websocket: any, context: any) => {
          try {
            return context;
          } catch (error) {
            console.error('WebSocket auth error:', error);
            throw new Error('Invalid authentication token');
          }
        },
      },
    },
 ```
 
 getting the error 
 
 `"error": "Could not connect to websocket endpoint ws://localhost:8000/graphql. Please check if the endpoint url is correct."`
 
 if we are using 
 
 ```
 subscriptions: {
      'subscriptions-transport-ws': {
        path: '/graphql', 
        onConnect: (connectionParams: any, websocket: any, context: any) => {
          try {
            return context;
          } catch (error) {
            console.error('WebSocket auth error:', error);
            throw new Error('Invalid authentication token');
          }
        },
      },
    },
```

It's working fine even though we haven't installed `subscriptions-transport-ws`

**Comment by enisdenjo** [comment]
Hi there, I dont seem to recognize this code aside from `onConnect` (whose usage is incorrect also). I need a lot more information if you want me to try helping you out. What is this? Can you share the full configuration? Where did you get this setup?
